packages/lesswrong/components/ea-forum/onboarding/EAOnboardingThankYouStage.tsx

```diff
@@ -86,18 +86,18 @@ export const EAOnboardingThankYouStage = ({classes}: {
   classes: ClassesType<typeof styles>,
 }) => {
   const {captureEvent} = useTracking();
-  const {goToNextStage, currentUser, updateCurrentUser} = useEAOnboarding();
+  const {currentStage, goToNextStage, currentUser, updateCurrentUser} = useEAOnboarding();
   const [subscribed, setSubscribed] = useState(true);
 
   useEffect(() => {
     // Default to subscribing to the digest
-    if (!currentUser.subscribedToDigest) {
+    if (!currentUser.subscribedToDigest && currentStage === 'thankyou') {
```

Comment:
Given the name of this component I sure am confused how this does anything. (Will check out the code and review later.)